internal/pkg/agent/install/user_windows_test.go

```diff
@@ -0,0 +1,65 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+//go:build windows
+
+package install
+
+import (
+	"strings"
+	"testing"
+)
+
+// TestRandomPassword tries to ensure that generated passwords meet the windows security constraints.
+func TestRandomPassword(t *testing.T) {
+	const testTimes = 100000
+	passwords := make(map[string]struct{})
+	for i := 0; i < testTimes; i++ {
+		password, err := RandomPassword()
+		if err != nil {
+			t.Fatalf("RandomPassword() returned an error: %v", err)
+		}
+
+		length := len(password)
+		if length < passwordMinLength || length > passwordMaxLength {
+			t.Fatalf("password %q is not within the allowed length range", password)
+		}
+
+		hasLower := false
+		hasUpper := false
+		hasDigit := false
+		hasSpecial := false
+
+		for _, char := range password {
+			switch {
+			case strings.ContainsRune(passwordCharsLower, char):
+				hasLower = true
+			case strings.ContainsRune(passwordCharsUpper, char):
+				hasUpper = true
+			case strings.ContainsRune(passwordCharsDigits, char):
+				hasDigit = true
+			case strings.ContainsRune(passwordCharsSpecial, char):
+				hasSpecial = true
+			default:
+				t.Fatalf("password %q contains an invalid character %q", password, string(char))
+			}
+		}
+
+		if !hasLower || !hasUpper || !hasDigit || !hasSpecial {
+			t.Fatalf("password %q does not contain all required character categories", password)
```

Comment:
Can you add the values for `hasLower`, `hasUpper`, `hasDigit`, and `hasSpecial` to the error.  If it ever does fail it will make it easier on the person looking at the CI failure.